fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/impl/registry/sync/packet/RegistrySyncPayload.java

```diff
@@ -17,75 +17,109 @@
 import it.unimi.dsi.fastutil.objects.Object2IntLinkedOpenHashMap;
 import it.unimi.dsi.fastutil.objects.Object2IntMap;
-import org.jetbrains.annotations.Nullable;
 
 import net.minecraft.network.PacketByteBuf;
 import net.minecraft.network.codec.PacketCodec;
 import net.minecraft.network.packet.CustomPayload;
-import net.minecraft.registry.Registries;
 import net.minecraft.registry.Registry;
+import net.minecraft.registry.RegistryKey;
 import net.minecraft.util.Identifier;
 
 import net.fabricmc.fabric.api.event.registry.RegistryAttribute;
 import net.fabricmc.fabric.api.event.registry.RegistryAttributeHolder;
-import net.fabricmc.fabric.api.networking.v1.PacketByteBufs;
+import net.fabricmc.fabric.impl.registry.sync.RegistryAttributeImpl;
 
 /**
  * A more optimized method to sync registry ids to client.
- * Produce smaller packet than old nbt-based method.
+ * Produces smaller packets than the old nbt-based method.
  *
- * <p>This method optimize the packet in multiple way:
+ * <p>This method optimizes the packet in multiple ways:
  * <ul>
- *     <li>Directly write into the buffer instead of using an nbt;</li>
- *     <li>Group all {@link Identifier} with same namespace together and only send those unique namespaces once for each group;</li>
- *     <li>Group consecutive rawIds together and only send the difference of the first rawId and the last rawId of the bulk before.
- *     This is based on the assumption that mods generally register all of their object at once,
+ *     <li>Directly writes into the buffer instead of using an nbt;</li>
+ *     <li>Groups all {@link Identifier}s with same namespace together and only sends those unique namespaces once for each group;</li>
+ *     <li>Groups consecutive rawIds together and only sends the difference of the first rawId and the last rawId of the bulk before.
+ *     This is based on the assumption that mods generally register all of their objects at once,
  *     therefore making the rawIds somewhat densely packed.</li>
  * </ul>
- *
- * <p>This method also split into multiple packets if it exceeds the limit, defaults to 1 MB.
  */
-public class DirectRegistryPacketHandler extends RegistryPacketHandler<DirectRegistryPacketHandler.Payload> {
-	/**
-	 * @see net.minecraft.network.packet.s2c.play.CustomPayloadS2CPacket#MAX_PAYLOAD_SIZE
-	 */
-	@SuppressWarnings("JavadocReference")
-	private static final int MAX_PAYLOAD_SIZE = Integer.getInteger("fabric.registry.direct.maxPayloadSize", 0x100000);
+public record RegistrySyncPayload(
+		Map<Identifier, Object2IntMap<Identifier>> registryMap,
+		Map<Identifier, EnumSet<RegistryAttribute>> registryAttributes
+) implements CustomPayload {
+	public static CustomPayload.Id<RegistrySyncPayload> ID = new CustomPayload.Id<>(Identifier.of("fabric", "registry/sync"));
+	public static PacketCodec<PacketByteBuf, RegistrySyncPayload> CODEC = CustomPayload.codecOf(RegistrySyncPayload::write, RegistrySyncPayload::read);
```

Comment:
```suggestion
	public static final CustomPayload.Id<RegistrySyncPayload> ID = new CustomPayload.Id<>(Identifier.of("fabric", "registry/sync"));
	public static final PacketCodec<PacketByteBuf, RegistrySyncPayload> CODEC = CustomPayload.codecOf(RegistrySyncPayload::write, RegistrySyncPayload::read);
```
I don't think anything modifies these

Reply:
It's in a record, `final` is implied.

Reply:
Wait wtf, how do I unsend a message